client/client_api.go

```diff
@@ -48,6 +62,13 @@ type GenericClient interface {
 	// TODO: return as list of config items
 	GetConfig(dsts ...interface{}) error
 
+	// GetFilteredConfig retrieves current config into dsts according to the provided filter.
+	GetFilteredConfig(filter Filter, dsts ...interface{}) error
+
+	UpdateConfig(ctx context.Context, items UpdateItems) (*generic.SetConfigResponse, error)
+
+	DeleteConfig(ctx context.Context, items UpdateItems) (*generic.SetConfigResponse, error)
```

Comment:
These are not much of an user-friendly improvement for the original gRPC service definition.. it's practically almost same if the user uses this methods or the generated gRPC [`generic.ManagerServiceClient`](https://github.com/ligato/vpp-agent/blob/master/proto/ligato/generic/manager_grpc.pb.go#L24). And it's actually worse for user to use this over the generated client..

My top concerns about this are:
- field `OverwriteAll` in the `UpdateItems` struct
  - introduces another name for the same operation: Resync/Replace/OverwriteAll (it is unfortuante that different parts of code use different name for this, but let's stick with single one per entity)
  - this option does not make sense in `DeleteConfig` method (replace and delete are mutually exclusive operations), instead it should be a parameter in the `UpdateConfig` method or simply as another method.. 
- type `generic.SetConfigResponse` used as return value in the methods of client interface
  - this forces non-gRPC implementation (localclient) to return this gRPC response wrapper type (currently returns `nil` though)
  - this should return slice of update results
- `UpdateItems` contains list of proto messages (items) and labels
  - this makes it impossible to apply labels per item in a single update even though the actual gRPC API allows this (rpc `SetConfig` containss list of `UpdateItem`s - _item + labels_ 
- the client interface is still missing universal method for getting entire config as return value (list of items)
- I think the `Filter` should only accept a single way to filter especially when filtering by IDs
  - user either wants to get specific items by referring to their ID 
  - or he wants to get all items that match the requested labels
  - allowing both filters in one request might be confusing about what we expect server to return